### PR TITLE
Fix up nullability and form schema handling in OpenAPI

### LINKED ISSUE #56903: Schemas for parameters should not specify `nullable: true`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The OpenAPI document generation support in .NET 9 preview 6 will produce parameter schemas with `nullable: true` if the endpoint parameter is a nullable type (either value or reference). This is incorrect because passing "null" for a query or header or route parameter does not set the parameter to `null` but rather sets it to the string "null".

### Expected Behavior

Don't specify `nullable: true` in the schema of query, header, or route parameters.

### Steps To Reproduce

Not minimal, but easy. Clone my [aspnet-openapi-examples]() project and run `dotnet build`, then look at the "q", "myCustomHeader", and "maybeValue" parameters for the Post on "/operations/{id}/parameters".

### Exceptions (if any)

_No response_

### .NET Version

9.0.100-preview.7.24365.15

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #56903.

Mark all parameters sourced from the query, header, route, and form from being marked as nullable.

Cloes https://github.com/dotnet/aspnetcore/issues/57082.

Attributes on the parameter associated with a `[FromForm]` attribute now get applied correctly because we pass in `ApiParameterDescription` when we generate the schema.

Addresses https://github.com/dotnet/aspnetcore/issues/57083.

Remove the `encoding` field on form parameters until we figure out the correct values to set.

### CODE REVIEW

File: src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs [hunk 1]
```diff
@@ -317,6 +318,12 @@ internal static void ApplyParameterInfo(this JsonNode schema, ApiParameterDescri
         {
             schema.ApplyRouteConstraints(constraints);
         }
+        // Parameters sourced from the header, query, and route parameter cannot be nullable.
+        if (parameterDescription.Source is { } bindingSource &&
+            (bindingSource == BindingSource.Header || bindingSource == BindingSource.Query || bindingSource == BindingSource.Path))
```
Comment (mikekistler) [inline src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:323]: Should form params also be excluded from being nullable?

Comment (captainsafia) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:323]: Looked into this. It appears that we'll want arguments bound from a form to also be excluded. I'll update the implementation and add tests.

**Review by mikekistler — APPROVED** [review]
Looks good! 👍 

I left a few suggestions on the tests which may or may not be good ideas.

File: src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Services/OpenApiSchemaService/OpenApiSchemaService.RequestBodySchemas.cs [hunk 2]
```diff
@@ -60,14 +60,23 @@ public async Task GetOpenApiRequestBody_GeneratesSchemaForPoco()
         });
     }
 
-    [Fact]
-    public async Task GetOpenApiRequestBody_GeneratesSchemaForPoco_WithValidationAttributes()
+    [Theory]
+    [InlineData(false, "application/json")]
+    [InlineData(true, "application/x-www-form-urlencoded")]
```
[the remark this answers is not in the document]
Comment (captainsafia) [reply src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Services/OpenApiSchemaService/OpenApiSchemaService.RequestBodySchemas.cs:65]: Good call -- added a test here.

**Pull request merged**